packages/@lightningjs/ui-components/src/mixins/withThemeStyles/index.js

```diff
@@ -64,21 +64,48 @@ export default function withThemeStyles(Base, mixinStyle) {
       });
     }
 
+    /**
+     * Climb the prototype chain to establish what component's extension rules this component should also inherit
+     * @returns {set} // set of strings
+     */
+    get _prototypeChain() {
+      if (this.__prototypeChain) return this.__prototypeChain;
+      const prototypeChain = new Set();
+      let proto = this;
+      do {
+        proto = Object.getPrototypeOf(proto);
+        if (null !== proto && typeof proto === 'object') {
+          try {
+            if (proto.constructor.__componentName)
+              prototypeChain.add(proto.constructor.__componentName);
+          } catch (error) {
+            // Catch error when __componentName is not set in Base component
```

Comment:
would it be worth logging something here, or is this just catching a normal case that isn't an error?

Reply:
Added context.error so it doesn't fail silently.